utils/lit/lit/TestRunner.py

```diff
@@ -471,6 +473,18 @@ def replace_line_number(match):
             # END commands are only honored if the rest of the line is empty.
             if not ln.strip():
                 break
+        elif command_type.startswith('COND_RUN('):
+            cond_run_requires_str = command_type[9:len(command_type)-1]
+            cond_run_requires = cond_run_requires_str.split(',')
+            # Trim spaces.
+            cond_run_requires = [require.strip()
+                      for require in cond_run_requires]
+            # Collapse lines with trailing '\\'.
+            if cond_script and cond_script[-1][1][-1] == '\\':
+                cond_script[-1] = cond_script[-1][1][:-1] + ln
+            else:
+                cond_run = (cond_run_requires, ln)
+                cond_script.append(cond_run)
```

Comment:
Shouldn't conditional run lines be ordered in the same list as normal run lines, rather than gathered separately and tacked onto the end?  Can these conditions be tested within the code gathering the lines, then added to the same script if they pass?  Then the `if command_type == 'RUN':` block could be modified to handle `COND_RUN(` at the same time, only checking the condition and potentially skipping if it's `command_type.startswith('COND_RUN(')`.

Reply:
Good catch.
Will fix it.

Reply:
Fixed.